rskj-core/src/test/java/co/rsk/net/NodeBlockProcessorTest.java

```diff
@@ -1082,4 +1084,66 @@ void failIfProcessBlockHeadersRequestCountHigher()  {
         verify(sender, never()).sendMessage(any());
 
     }
+
+    @Test
+    void processNewBlockHashesMessageOnlyRequestsNonAdvancedBlock() {
+        final NetBlockStore store = new NetBlockStore();
+        final Peer sender = mock(SimplePeer.class);
+        final BlockNodeInformation nodeInformation = mock(BlockNodeInformation.class);
+        final SyncConfiguration syncConfiguration = SyncConfiguration.IMMEDIATE_FOR_TESTING;
+        final Blockchain blockchain = new BlockChainBuilder().ofSize(0);
+
+        TestSystemProperties config = new TestSystemProperties();
+        BlockSyncService blockSyncService = new BlockSyncService(config, store, blockchain, nodeInformation, syncConfiguration, DummyBlockValidator.VALID_RESULT_INSTANCE);
+        final NodeBlockProcessor processor = new NodeBlockProcessor(store, blockchain, nodeInformation, blockSyncService, syncConfiguration);
```

Comment:
(minor thing) I'd extract this logic to a method as it seems the same code is used in other tests just to create the processor.